ledger-core/v2/vanilla/reflectkit/func_code.go

```diff
@@ -50,3 +50,23 @@ func unwrapIface(v interface{}) (word unsafe.Pointer, kind uint8) {
 	iface := (*emptyInterface)(unsafe.Pointer(&v))
 	return iface.word, iface.typ.kind
 }
+
+// A faster version (25-35%) to check for interface nil
+func IsNil(v interface{}) bool {
+	if v == nil {
+		return true
+	}
+	ptr, kind := unwrapIface(v)
+	return ptr == nil && kind&kindDirectIface != 0
```

Comment:
This is really bad code because it relies on assumptions that are not part of spec. It could break after random version change (even minor). Do we really need it?

Reply:
1) such change won't be occasional, like with major version that will be carefully evaluated anyway
2) there are tests to ensure behavior on possible cases
3) there is an available fallback option - so even if the code will stop working, it can be replaced without impact to users